src/PackagePrivate/InMemoryEntityTermStore.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\TermStore\MediaWiki\PackagePrivate;
+
+use Wikibase\DataModel\Entity\EntityId;
+
+/**
+ * in-memory, @inheritDoc
+ */
+class InMemoryEntityTermStore implements EntityTermStore {
+	private $entityTerms = [];
+
+	/**
+	 * @inheritDoc
+	 */
+	public function setTerms( EntityId $entityId, array $termsArray ) {
+		$this->entityTerms[$entityId->getSerialization()] = $termsArray;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function unsetTerms( EntityId $entityId ) {
+		unset( $this->entityTerms[$entityId->getSerialization()] );
+	}
+
+	/**
+	 * check that terms for the given entity id has the given terms stored
+	 *
+	 * @param EntityId $entityId
+	 * @param array $termsArray same as $termsArray for InMemoryEntityTermStore::setTerms()
+	 *
+	 * @return bool
+	 */
+	public function hasTerms( EntityId $entityId, array $termsArray ) {
+		if ( !isset( $this->entityTerms[$entityId->getSerialization()] ) ) {
+			return false;
+		}
+
+		return $termsArray === $this->entityTerms[$entityId->getSerialization()];
+	}
+
+	/**
+	 * Check that given entity id has no terms stored
+	 */
+	public function hasNoTerms( EntityId $entityId ) {
```

Comment:
The name implies that this is `!self::hasTerms()`… I guess `hasAnyTerms( EntityId $entityId )` (which can of course be negated) would be more fitting.

Reply:
Not sure about this honestly .. if I would name something for `!self::hasTerms` I would go for `self::doesNotHaveTerms`.. usually in English `has no doors` implies `does not have any doors at all` rather than `does not have (certain) doors`.. though I might be wrong, some native English might help here @tarrow :)

Well even if I'm not wrong, this could still be confusing .. will think of a better name.

Reply:
@mariushoch 
I'm thinking of one of two options:
- go with your suggestion `hasAnyTerms`
- rename `hasTerms` only to `hasExactTerms`
not sure which one is better

Reply:
@mariushoch I went with the second option just because it is less work .. let me know if it looks better from confusing function names POV

Reply:
name bikeshedding, I love them!

how about:
- isEmpty
- containsTerms (or something along these lines, containsGivenTerms, containsAllOfTermForEntity, etc)

Reply:
haha bikeshedding is the best anyway.. I think `contains` is more problematic so I'd stay with `has`.

Contains will imply that you'll recursively check existing terms to see if given terms are in there, which is a functionality we do not need yet. You can do `containsExcactly` but then it is the same as `hasExactTerms` or?